File: mpp-firma-transport-c#/FirmaService/IFirmaServices.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using FirmaModel1;
using Utils;



namespace FirmaService
{
    public interface IFirmaServices
    {
        Optional<Oficiu> login(String username, String password, IFirmaObserver client) ;
        void logout(String username) ;
        List<CursaDTO> getAllCurse();

        List<CursaDTO> getCursaByDestinatieData(String destinatie, String data) ;

        List<LocDTO> getLocuriDisponibile(long idCursa) ;
        void makeReservation(long idCursa, String numeClient, int nrLocuri) ;

        RezervareDTO getRezervareById(long rezervare) ;


    }
}

File: MPP-firma-transport-java/FirmaService/src/main/java/org/example/IFirmaServices.java
package org.example;

import java.util.List;
import java.util.Optional;

public interface IFirmaServices {

    Optional<Oficiu> login(String username, String password, IFirmaObserver client) throws AppException;
    void logout(String username) throws AppException;
    List<CursaDTO> getAllCurse() throws AppException;

    List<CursaDTO> getCursaByDestinatieData(String destinatie, String data) throws AppException;

    List<LocDTO> getLocuriDisponibile(Long idCursa) throws AppException;
    void makeReservation(Long idCursa,String numeClient, Integer nrLocuri) throws AppException;

    RezervareDTO getRezervareById(Long rezervare) throws AppException;
}
